### PR TITLE
Fix some tests

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

test_distributed_ddl - https://s3.amazonaws.com/clickhouse-test-reports/0/fed6f50da6035c53550ab84aba9055a4ccd83788/integration_tests__tsan__[1/4].html
test_drop_replica - https://s3.amazonaws.com/clickhouse-test-reports/0/a307e0d95db7afcf8df13870456d8fc624308081/integration_tests__asan__[3/3].html
test_system_replicated_fetches - https://s3.amazonaws.com/clickhouse-test-reports/0/98b3617ce2c63625e617cc2dadf64b7e9769228a/integration_tests__asan__[1/3].html

Also it will suppress https://github.com/ClickHouse/ClickHouse/issues/48181

### CODE REVIEW

File: src/Interpreters/DDLWorker.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include <Common/ThreadPool.h>
 #include <Common/ZooKeeper/IKeeper.h>
 #include <Storages/IStorage_fwd.h>
+#include <Storages/MergeTree/ZooKeeperRetries.h>
```
Comment (antonio2368) [inline src/Interpreters/DDLWorker.h:9]:
```suggestion
```

File: src/Interpreters/executeDDLQueryOnCluster.cpp [hunk 2]
```diff
@@ -37,6 +37,27 @@ namespace ErrorCodes
...
+static RetriesForDDL getRetriesForDistributedDDL()
+{
+    const auto & config_ref = Context::getGlobalContextInstance()->getConfigRef();
+    auto info = ZooKeeperRetriesInfo(
+        "DistributedDDL",
+        &Poco::Logger::get("DDLQueryStatusSource"),
+        config_ref.getInt("distributed_ddl_keeper_max_retries", 5),
+        config_ref.getInt("distributed_ddl_keeper_initial_backoff_ms", 100),
+        config_ref.getInt("distributed_ddl_keeper_max_backoff_ms", 5000)
+        );
```
Comment (antonio2368) [inline src/Interpreters/executeDDLQueryOnCluster.cpp:55]:
```suggestion
    );
```

File: src/Interpreters/executeDDLQueryOnCluster.cpp [hunk 3]
```diff
@@ -37,6 +37,27 @@ namespace ErrorCodes
...
+
+    auto ctl = ZooKeeperRetriesControl("executeDDLQueryOnCluster", info);
+    return {info, ctl};
```
Comment (antonio2368) [inline src/Interpreters/executeDDLQueryOnCluster.cpp:58]:
```suggestion
    return {std::move(info), std::move(ctl)};
```

File: src/Interpreters/executeDDLQueryOnCluster.h [hunk 4]
```diff
@@ -43,7 +44,7 @@ struct DDLQueryOnClusterParams
 BlockIO executeDDLQueryOnCluster(const ASTPtr & query_ptr, ContextPtr context, const DDLQueryOnClusterParams & params = {});
 
 BlockIO getDistributedDDLStatus(
-    const String & node_path, const DDLLogEntry & entry, ContextPtr context, const std::optional<Strings> & hosts_to_wait = {});
+    const String & node_path, const DDLLogEntry & entry, ContextPtr context, const std::optional<Strings> & hosts_to_wait);
```
Comment (antonio2368) [inline src/Interpreters/executeDDLQueryOnCluster.h:47]: Not related that much to PR but `const std::optional<Strings> & ` is bad, better do `const Strings * hosts_to_wait` so you don't have to copy to `std::optional` if you have list of hosts.

**Comment by tavplubix** [comment]
Stateless tests flaky check (asan) - Test runs too long - OK
Stress test (msan) - we failed to check for long-running queries due to memory limit - https://github.com/ClickHouse/ClickHouse/pull/46089 will help

**Pull request merged**